utils/data-processing-initialization/src/main/java/com/github/cafdataprocessing/utilities/initialization/WorkflowRemover.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2015-2017 Hewlett Packard Enterprise Development LP.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.utilities.initialization;
+
+import com.github.cafdataprocessing.processing.service.client.ApiException;
+import com.github.cafdataprocessing.processing.service.client.api.ProcessingRulesApi;
+import com.github.cafdataprocessing.processing.service.client.api.WorkflowsApi;
+import com.github.cafdataprocessing.processing.service.client.model.ExistingProcessingRule;
+import com.github.cafdataprocessing.processing.service.client.model.ExistingWorkflow;
+import com.github.cafdataprocessing.processing.service.client.model.ExistingWorkflows;
+import com.github.cafdataprocessing.processing.service.client.model.ProcessingRules;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Removes processing workflows (including its components) based on project ID and
+ * matching names in preparation for creating new workflows.
+ */
+public class WorkflowRemover {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WorkflowRemover.class);
+
+    /**
+     * Removes any existing processing workflows (including processing rules etc under the workflow) under the specified projectId
+     * that match the name provided.
+     * @param apisProvider Provides access to processing service APIs so retrieval and delete requests may be sent via
+     *                     the appropriate API.
+     * @param projectId ProjectId that checked workflows should be under.
+     * @param workflowNameToRemove If any workflows have a name that matches this value they will be removed.
+     * @throws ApiException If an error occurs contacting the processing API.
+     */
+    public static void removeMatchingWorkflows(ProcessingApisProvider apisProvider, String projectId,
```

Comment:
Run a static code analysis to check for line length, use of "final" qualifier on parameters and local variables, etc.